code/clue.js

```diff
@@ -7,53 +7,267 @@ const mrGreen = {
+  bat,
+  trophy,
+  pistol
+];
 
-const rooms = []
+const rooms = [
+  'Dining Room',
+  'Conservatory',
+  'Kitchen',
+  'Study',
+  'Library',
+  'Billiard Room',
+  'Lounge',
+  'Ballroom',
+  'Hall',
+  'Spa',
+  'Living Room',
+  'Observatory',
+  'Theater',
+  'Guest House',
+  'Patio'
+];
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
+// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION:
 const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+  return array[Math.floor(Math.random() * array.length)];
+};
+
+// OBJECT THAT KEEPS THE MYSTERY:
+const mystery = {
+  // The values will be set later after running the pickKiller, pickWeapon and pickRoom functions
+};
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// PICK KILLER FUNCTION - This function will be invoked when you click on the killer card.
+document.getElementById("killerCard").addEventListener("click", pickKiller);
 
-// This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+function pickKiller() {  // Shows loading animation
+  document.getElementById('loader1').style.display = 'block';
+  setTimeout(delayedPickKiller, 1500); //Delays pickKiller function, calls it after 1.5sec
+};
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+const delayedPickKiller = () => {  // This will randomly select a killer from the suspects. And add that to the mystery object.
+  document.getElementById('loader1').style.display = 'none';
+  document.getElementById("killerCard").removeEventListener("click", pickKiller); //so that the card can't be clicked again and trigger the loader
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+  if(mystery.killer === undefined) {   //To check if card has been clicked already or not - undefined= not clicked
+    mystery.killer = randomSelector(suspects);
+    mystery.killer.favouriteWeapon = randomSelector(weapons);  //Randomly reassigns killer's favorite weapon
+      
+    // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+    document.getElementById('killerCard').style.background = mystery.killer.color;
```

Comment:
Is there a possible advantage to declaring these elements as constants?
`const killerCardElement = document.getElementById('killerCard')`